File: models/EntrepriseModel.js
const bcrypt = require("bcryptjs");
const mongoose = require("mongoose");
const { secteursActivites, statusPACKS } = require("../utils/FormatApi");

const EntrepriseSchema = new mongoose.Schema({
  username: {
    type: String,
    require: false,
    unique: true
  },
  full_name: {
    type: String,
    required: false
  },
  firstname: {
    type: String,
    required: false
  },
  lastname: {
    type: String,
    required: false
  },
  title_post: {
    type: String,
    required: false
  },
  email: {
    type: String,
    required: false,
    unique:true
  },
  employers_count: {
    type: String, require: false
  },
  dateNaissance: {
    type: String, require: false
  },
  dateNaissance_entreprise: {
    type: String, require: false
  },
  telephone: {
    type: String,
    required: false
  },
  telephone_entreprise: {//ok
    type: String,
    required: false
  },
  adresse: {
    type: String,
    required: false
  },
  description_entreprise: {//ok
    type: String
  },
  logo: { //ok
    type: String
  },
  pays_entreprise: {//ok
    type: String,
  },
  addresse_entreprise: {
    type: String
  },
  salaire_capital: {//of
    type: String,
    required: false
  },
  typeStatut: { type: String, default: "STARTER", enum: ["PREMIUM", "NORMAL", "STARTER"] },
  comptePaiement: {
    solde: {
      type: Number,
      default: 0
    },
    dateDernierPaiement: {
      type: Date,
      default: null
    },
    moyenPaiement: {
      type: String
    }
  },
  maps: {
    type: String, default: false
  },
  maps_entreprise: {
    type: String, default: false, default: "#"
  },
  offres: [
    { type: Object }
  ],
  secteur_activites: [{ type: Object }],
  langues: [{ type: Object }],
  site_web: {
    type: String, default: "#", require: false
  },
  facebook_url: {
    type: String, default: "#", require: false
  },
  twitter_url: {
    type: String, default: "#", require: false
  },
  linkedin_url: {
    type: String, default: "#", require: false
  },
  instagram_url: {
    type: String, default: "#", require: false
  },

  password: {
    type: String,
    required: false
  },


  // packs pour les entreprise
  account: {
    solde: { type: Number, default: 0 },
    pack: { type: String, default: statusPACKS[0] },
    dateNow: { type: String, default: Date.now },
    dateEnd: { type: String },
    count_sms: { type: Number, default: 0 }
  },
  token: {
    type: String, required: false
  },
  is_active: {
    type: Boolean,
    default: false
  },
  access: {
    type: Boolean,
    default: true
  },
});

const EntrepriseModel = mongoose.model("entreprise", EntrepriseSchema);



const listAdminList = [
  {
    "username": "utilisateur123",
    "full_name": "John Doe",
    "firstname": "John",
    "lastname": "Doe",
    "title_post": "Ingénieur logiciel",
    "email": "john.doe@email.com",
    "email_entreprise": "contact@entreprise.com",
    "employers_count": "50",
    "dateNaissance": "01/01/1980",
    "dateNaissance_entreprise": "01/01/1990",
    "telephone": "123-456-7890",
    "telephone_entreprise": "987-654-3210",
    "adresse": "123 Rue de la Rue",
    "description_entreprise": "Une entreprise innovante",
    "logo": "url_du_logo.jpg",
    "pays_entreprise": "France",
    "addresse_entreprise": "456 Avenue de l'Entreprise",
    "salaire_capital": "1000000",
    "typeStatut": "PREMIUM",
    "comptePaiement": {
      "solde": 5000,
      "dateDernierPaiement": "2023-09-15",
      "moyenPaiement": "Carte de crédit"
    },
    "maps": "url_de_la_carte",
    "maps_entreprise": "url_de_la_carte_entreprise",
    "offres": [],
    "secteur_activites": [
      { label: "Technologie de l'information", value: "technologie_information" },
      { label: "Développement informatique", value: "developpement_informatique" },
    ],
    "langues": [
      { label: "Technologie de l'information", value: "technologie_information" },
      { label: "Développement informatique", value: "developpement_informatique" },
    ],
    "site_web": "https://www.monentreprise.com",
    "facebook_url": "https://www.facebook.com/monentreprise",
    "twitter_url": "https://www.twitter.com/monentreprise",
    "linkedin_url": "https://www.linkedin.com/monentreprise",
    "instagram_url": "https://www.instagram.com/monentreprise",
    "password": "motdepasse123",
    "token": "jeton_d'authentification",
    "is_active": true,
    "access": true
  }
]




// Fonction pour créer un nouvel administrateur par default
// const createAdminDefault = async (itemAmdin) => {
//   const newadmin = new EntrepriseModel(itemAmdin);
//   const hashedPassword = await bcrypt.hash(itemAmdin.password, 10);
//   newadmin.password = hashedPassword;
//   await newadmin.save()
//     .then((savedAdmin) => {
//       console.log('Nouvelle Entrperise créer avec succès.');
//     })
//     .catch((error) => {
//       console.log("Impossible de créer Entreprise : ");
//     });
// }

// create administrateur
// for (var item of listAdminList) {
//   createAdminDefault(item);
// }








module.exports = EntrepriseModel;

File: lib/src/model/EntrepriseModel.dart


class EnterpriseModel {
  String? username;
  String? fullName;
  String? firstName;
  String? lastName;
  String? titlePost;
  String? email;
  String? employersCount;
  String? dateNaissance;
  String? dateNaissanceEnterprise;
  String? telephone;
  String? telephoneEnterprise;
  String? address;
  String? enterpriseDescription;
  String? logo;
  String? enterpriseCountry;
  String? enterpriseAddress;
  String? capitalSalary;
  String typeStatus;
  String maps;
  String mapsEnterprise;
  String website;
  String facebookUrl;
  String twitterUrl;
  String linkedinUrl;
  String instagramUrl;
  String? token;
  bool isActive;
  bool access;

  EnterpriseModel({
    this.username,
    this.fullName,
    this.firstName,
    this.lastName,
    this.titlePost,
    this.email,
    this.employersCount,
    this.dateNaissance,
    this.dateNaissanceEnterprise,
    this.telephone,
    this.telephoneEnterprise,
    this.address,
    this.enterpriseDescription,
    this.logo,
    this.enterpriseCountry,
    this.enterpriseAddress,
    this.capitalSalary,
    this.typeStatus = "STARTER",
    this.maps = "false",
    this.mapsEnterprise = "#",
    this.website = "#",
    this.facebookUrl = "#",
    this.twitterUrl = "#",
    this.linkedinUrl = "#",
    this.instagramUrl = "#",
    this.token,
    this.isActive = false,
    this.access = true,
  });

  //Méthode pour désérialiser un JSON en un objet EnterpriseModel
  factory EnterpriseModel.fromJson(Map<String, dynamic> json) {
    return EnterpriseModel(
      username: json['username'],
      fullName: json['full_name'],
      firstName: json['firstname'],
      lastName: json['lastname'],
      titlePost: json['title_post'],
      email: json['email'],
      employersCount: json['employers_count'],
      dateNaissance: json['dateNaissance'],
      dateNaissanceEnterprise: json['dateNaissance_entreprise'],
      telephone: json['telephone'],
      telephoneEnterprise: json['telephone_entreprise'],
      address: json['adresse'],
      enterpriseDescription: json['description_entreprise'],
      logo: json['logo'],
      enterpriseCountry: json['pays_entreprise'],
      enterpriseAddress: json['addresse_entreprise'],
      capitalSalary: json['salaire_capital'],
      typeStatus: json['typeStatut'],
      maps: json['maps'],
      mapsEnterprise: json['maps_entreprise'],
      website: json['site_web'],
      facebookUrl: json['facebook_url'],
      twitterUrl: json['twitter_url'],
      linkedinUrl: json['linkedin_url'],
      instagramUrl: json['instagram_url'],
      token: json['token'],
      isActive: json['is_active'],
      access: json['access'],
    );
  }

  // Méthode pour sérialiser un objet EnterpriseModel en JSON
  Map<String, dynamic> toJson() {
    return {
      'username': username,
      'full_name': fullName,
      'firstname': firstName,
      'lastname': lastName,
      'title_post': titlePost,
      'email': email,
      'employers_count': employersCount,
      'dateNaissance': dateNaissance,
      'dateNaissance_entreprise': dateNaissanceEnterprise,
      'telephone': telephone,
      'telephone_entreprise': telephoneEnterprise,
      'adresse': address,
      'description_entreprise': enterpriseDescription,
      'logo': logo,
      'pays_entreprise': enterpriseCountry,
      'addresse_entreprise': enterpriseAddress,
      'salaire_capital': capitalSalary,
      'typeStatut': typeStatus,
      'maps': maps,
      'maps_entreprise': mapsEnterprise,
      'site_web': website,
      'facebook_url': facebookUrl,
      'twitter_url': twitterUrl,
      'linkedin_url': linkedinUrl,
      'instagram_url': instagramUrl,
      'token': token,
      'is_active': isActive,
      'access': access,
    };
  }




}
